src/wp-includes/blocks/template-part.php

```diff
@@ -0,0 +1,155 @@
+		if ( $template_part_post ) {
+			// A published post might already exist if this template part was customized elsewhere
+			// or if it's part of a customized template.
+			$content    = $template_part_post->post_content;
+			$area_terms = get_the_terms( $template_part_post, 'wp_template_part_area' );
+			if ( ! is_wp_error( $area_terms ) && false !== $area_terms ) {
+				$area = $area_terms[0]->name;
+			}
+		} else {
+			// Else, if the template part was provided by the active theme,
+			// render the corresponding file content.
+			$template_part_file_path = get_stylesheet_directory() . '/block-template-parts/' . $attributes['slug'] . '.html';
+			if ( 0 === validate_file( $attributes['slug'] ) && file_exists( $template_part_file_path ) ) {
+				$content = _inject_theme_attribute_in_block_template_content( file_get_contents( $template_part_file_path ) );
+			}
+		}
+	}
+
+	if ( is_null( $content ) && is_user_logged_in() ) {
+		if ( ! isset( $attributes['slug'] ) ) {
+			// If there is no slug this is a placeholder and we dont want to return any message.
+			return;
+		}
+		return sprintf(
+			/* translators: %s: Template part slug. */
+			__( 'Template part has been deleted or is unavailable: %s' ),
+			$attributes['slug']
+		);
+	}
+
+	if ( isset( $seen_ids[ $template_part_id ] ) ) {
+		// WP_DEBUG_DISPLAY must only be honored when WP_DEBUG. This precedent
+		// is set in `wp_debug_mode()`.
+		$is_debug = defined( 'WP_DEBUG' ) && WP_DEBUG &&
+			defined( 'WP_DEBUG_DISPLAY' ) && WP_DEBUG_DISPLAY;
+
+		return $is_debug ?
+			// translators: Visible only in the front end, this warning takes the place of a faulty block.
+			__( '[block rendering halted]' ) :
+			'';
+	}
+
+	// Run through the actions that are typically taken on the_content.
+	$seen_ids[ $template_part_id ] = true;
+	$content                       = do_blocks( $content );
+	unset( $seen_ids[ $template_part_id ] );
+	$content = wptexturize( $content );
+	$content = convert_smilies( $content );
+	$content = shortcode_unautop( $content );
+	if ( function_exists( 'wp_filter_content_tags' ) ) {
+		$content = wp_filter_content_tags( $content );
+	} else {
+		$content = wp_make_content_images_responsive( $content );
+	}
+	$content = do_shortcode( $content );
+
+	if ( empty( $attributes['tagName'] ) ) {
+		$defined_areas = get_allowed_block_template_part_areas();
+		$area_tag      = 'div';
+		foreach ( $defined_areas as $defined_area ) {
+			if ( $defined_area['area'] === $area && isset( $defined_area['area_tag'] ) ) {
+				$area_tag = $defined_area['area_tag'];
+			}
+		}
+		$html_tag = $area_tag;
+	} else {
+		$html_tag = esc_attr( $attributes['tagName'] );
+	}
+	$wrapper_attributes = get_block_wrapper_attributes();
+
+	return "<$html_tag $wrapper_attributes>" . str_replace( ']]>', ']]&gt;', $content ) . "</$html_tag>";
+}
+
+/**
+ * Returns an array of variation objects for the template part block.
+ *
+ * @return array Array containing the block variation objects.
+ */
```

Comment:
```suggestion
/**
 * Returns an array of variation objects for the template part block.
 *
 * @since 5.9.0
 *
 * @return array Array containing the block variation objects.
 */
```

Reply:
This will need to be changed in Gutenberg as this file is copied from `@wordpress/block-library`.

Reply:
See my other comment about `@since` versions for GB-sync'd files. To confirm, I've grepped `since` in GB's `block-library` package directory, and I've only found one match. So I think we generally don't seem to add `@since` to dynamic blocks PHP in `block-library`.